test/blockly.test.js

```diff
@@ -0,0 +1,144 @@
+const test = require('tape');
+
+const {markdownToRedacted, sourceAndRedactedToRestored} = require('./utils');
+
+const blockly = require('../src/blockly');
+
+test('blockly plugin can redact inline blockly', t => {
+  t.plan(1);
+  const markdown =
+    'Some simple markdown with <xml><block type="example"/></xml> a blockly block';
+  const expected =
+    'Some simple markdown with [blockly block][0] a blockly block\n';
+
+  const rendered = markdownToRedacted(markdown, blockly);
+  t.equal(rendered, expected);
+});
+
+test('blockly plugin can redact blockly on its own line', t => {
+  t.plan(1);
+  const markdown =
+    'Some simple markdown with\n' +
+    '\n' +
+    '<xml><block type="example"/></xml>\n' +
+    '\n' +
+    'a blockly block';
+  const expected =
+    'Some simple markdown with\n' +
+    '\n' +
+    '[blockly block][0]\n' +
+    '\n' +
+    'a blockly block\n';
+
+  const rendered = markdownToRedacted(markdown, blockly);
+  t.equal(rendered, expected);
+});
+
+test('blockly plugin can redact inline blockly that spans multiple lines', t => {
+  t.plan(1);
+  const markdown =
+    'Some simple markdown with <xml>\n' +
+    '  <block type="example">\n' +
+    '    <title name="complex"/>\n' +
+    '  </block>\n' +
+    '</xml> a blockly block';
+  const expected =
+    'Some simple markdown with [blockly block][0] a blockly block\n';
+
+  const rendered = markdownToRedacted(markdown, blockly);
+  t.equal(rendered, expected);
+});
+
+test('blockly plugin can redact block blockly that spans multiple lines', t => {
+  t.plan(1);
+  const markdown =
+    'Some simple markdown with\n' +
+    '\n' +
+    '<xml>\n' +
+    '\n' +
+    '  <block type="example"/>\n' +
+    '\n' +
+    '</xml>\n' +
+    '\n' +
+    'a blockly block';
+  const expected =
+    'Some simple markdown with\n' +
+    '\n' +
+    '[blockly block][0]\n' +
+    '\n' +
+    'a blockly block\n';
+
+  const rendered = markdownToRedacted(markdown, blockly);
+  t.equal(rendered, expected);
+});
+
+test('blockly plugin can NOT redact inline blockly blocks with excess newlines', t => {
+  t.plan(1);
+  // This is a product of the fact that in this situation, markdown parses
+  // this content as two separate blocks, and inline tokenizers can only
+  // tokenize within a single block.
```

Comment:
Good to call out! I assume this is fine, because ideally all content creators would have access to a "Preview" mode that uses the same renderer as the final output?

Reply:
Ideally, yes.

Although note that right now, we're not using a remark plugin for rendering blockly and are instead relying on browser XML for that, which _will_ support breaking up the XML over multiple blocks. Our eventual implementation of that plugin will likely have to address this inconsistency in our content.